### ISSUE TITLE
Permit the reconciler history to be cleared

### ISSUE THREAD

**Issue by grzm** [omcljs/om#784]
It's useful to be able to clear the reconciler history cache to reset the app. For example, if a user logs out, we may want to ensure that any accumulated state is removed.

**Comment by grzm** [comment]
This could be accomplished in a couple of ways. One is to just replace the reconciler's history with a new Cache instance. Another would be to implement a `clear` method on the Cache type itself.

```clojure
(deftype Cache [arr index size]
  Object
 ;; ...
  (clear [this]
    (reset! (.-index this) {})
    (set! (.-arr this) #js [])
    this))
```

**Comment by swannodette** [comment]
@grzm patch welcome - have you submitted your CA?

**Comment by grzm** [comment]
@swannodette I don't believe I've submitted one for om. My email address is on my profile, if you would send me an electronic CA request.

**Comment by swannodette** [comment]
History is going to be made configurable so this will be subsumed by that.
